src/github/pullRequestManager.ts

```diff
@@ -1904,11 +1905,30 @@ export class PullRequestManager implements vscode.Disposable {
 			base: `${pullRequest.base.repositoryCloneUrl.owner}:${encodeURIComponent(pullRequest.base.ref)}`,
 			head: `${pullRequest.head.repositoryCloneUrl.owner}:${encodeURIComponent(pullRequest.head.ref)}`
 		});
-
 		pullRequest.mergeBase = data.merge_base_commit.sha;
 
-		Logger.debug(`Fetch file changes and merge base of PR #${pullRequest.number} - done`, PullRequestManager.ID);
-		return data.files;
+		let files: Array<IRawFileChange> = [];
+		if (data.files.length >= MAX_FILE_CHANGES_IN_COMPARE_COMMITS) {
+			// compareCommits will return a maximum of 300 changed files
+			// If we have (maybe) more than that, we'll need to fetch them with listFiles API call
+			Logger.debug(`More than ${MAX_FILE_CHANGES_IN_COMPARE_COMMITS} files changed, fetching all file changes of PR #${pullRequest.number}`, PullRequestManager.ID);
+			// @ts-ignore
+			files = await octokit.paginate(`GET /repos/:owner/:repo/pulls/:pull_number/files`, {
+				owner: pullRequest.base.repositoryCloneUrl.owner,
+				pull_number: pullRequest.number,
+				repo: remote.repositoryName,
+				per_page: 100
+			}, (r) => {
+				Logger.debug(`Fetched another batch of ${r.data.length} file changes for PR #${pullRequest.number}`, PullRequestManager.ID);
+				return r;

```

Comment:
I think this should be `r.data`

Reply:
I don't think it should. Pagination function handles selecting the data it seems, this last optional parameter _can_ be used for transforming the data, but I only wanted to log some debug info so I could be happy it was respecting the batch size I specified. 

Anyway, it works as expected returning r and not r.data.

Reply:
On second thought, I just removed the logging if that's ok.
I don't think there's any great reason to keep it, I just put it in to prove the pagination was happening during testing.